src/lib/support/tests/TestScoped.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/support/Scoped.h>
+#include <lib/support/UnitTestRegistration.h>
+
+#include <nlunit-test.h>
+#include <string.h>
+
+namespace {
+
+using namespace chip;
+
+void TestScopedVariableChange(nlTestSuite * inSuite, void * inContext)
+{
+    int x = 123;
+
+    {
+        ScopedChange<int> change1(x, 10);
```

Comment:
Are the explicit type annotations needed here?  I wouldn't think so, and would think our "here is how you use this" example (this test) would not use them, to avoid people doing it the "wrong" way elsewhere.

Reply:
Indeed not needed. Updated.